### PR TITLE
support Symbol objects

### PR DESCRIPTION
**Pull request by davidchambers**
This pull request changes the way Symbol objects are shown:

| Expression                | Previously gave…      | Now gives…            |
|:------------------------- |:--------------------- |:--------------------- |
| `show (Symbol.iterator)`  | `'<object Symbol>'`   | `'Symbol.iterator'`   |
| `show (Symbol ())`        | `'<object Symbol>'`   | `'Symbol ()'`         |
| `show (Symbol ('x'))`     | `'<object Symbol>'`   | `'Symbol ("x")'`      |
| `show (Symbol.for ('x'))` | `'<object Symbol>'`   | `'Symbol ("x")'` :\   |

Additionally, `Array#@@show` and `Object#@@show` now include symbol properties (see [`Object.getOwnPropertySymbols`][1]).

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertySymbols

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,11 +38,32 @@ const $$show = '@@show';
...
+  Symbol.match,
+  Symbol.matchAll,
+  Symbol.replace,
+  Symbol.search,
+  Symbol.species,
+  Symbol.split,
+  Symbol.toPrimitive,
+  Symbol.toStringTag,
+  Symbol.unscopables,
+].filter (x => typeof x === 'symbol');
```
Comment (Avaq) [inline index.js]: I think this could be a Set

File: index.js [hunk 2]
```diff
@@ -114,6 +135,11 @@ const show = x => {
         'new String (' + show (x.valueOf ()) + ')' :
         JSON.stringify (x);
 
+    case '[object Symbol]':
+      for (const s of wellKnownSymbols) if (s === x) return x.description;
```
Comment (Avaq) [inline index.js]: And then this is just `if (wellKnownSymbols.has(x)) return x.description`. Performance should remain the same.

File: test/index.js [hunk 3]
```diff
@@ -166,3 +166,47 @@ test ('custom @@show method (non-Object object)', () => {
...
+  eq (show ({[Symbol ('x')]: 0, [Symbol ('x')]: 0}), '{[Symbol ("x")]: 0, [Symbol ("x")]: 0}');
+  eq (show ({x: 0, [Symbol ('x')]: 0}), '{"x": 0, [Symbol ("x")]: 0}');
+});
+
+test ('array property symbols', () => {
+  const xs = ['foo', 'bar', 'baz'];
+  xs.z = 0;
+  xs[Symbol ('y')] = 0;
+  xs[Symbol ('x')] = 0;
+  eq (show (xs), '["foo", "bar", "baz", "z": 0, [Symbol ("x")]: 0, [Symbol ("y")]: 0]');
```
Comment (Avaq) [inline test/index.js:211]: This representation doesn't evaluate, though. An alternative would be to output `Object.assign(["foo", "bar", "baz"], {"z": 0, [Symbol ("x")]: 0, [Symbol ("y")]: 0})` or something. Maybe not worth it?

**Review by Avaq — APPROVED** [review]

File: test/index.js (hunk 3 above)
Comment (davidchambers) [reply test/index.js:211]: You're quite right. This is, though, consistent with existing behaviour:

```javascript
> show (/x/.exec ('xyz'))
'["x", "groups": undefined, "index": 0, "input": "xyz"]'
```

We could change the representation of non-index properties, but that would be an orthogonal change.

Comment (Avaq) [reply test/index.js:211]: Sure! Feel free to merge this PR by the way, I just wanted to note it. I'm not sure if it warrants an issue and the consequent work. I can already see the edge cases now, so maybe not open this can. :p
